File: app/src/main/java/com/djrausch/billtracker/adapters/PaidDateRecyclerViewAdapter.java
package com.djrausch.billtracker.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.djrausch.billtracker.R;
import com.djrausch.billtracker.models.BillPaid;

import org.joda.time.DateTime;

import io.realm.OrderedRealmCollection;
import io.realm.RealmRecyclerViewAdapter;

/**
 * Created by white on 7/25/2016.
 */
public class PaidDateRecyclerViewAdapter extends RealmRecyclerViewAdapter<BillPaid, PaidDateRecyclerViewAdapter.ViewHolder> {


    public PaidDateRecyclerViewAdapter(Context context, OrderedRealmCollection<BillPaid> data) {
        super(context, data, true);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_paid, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        BillPaid billPaid = getData().get(position);

        holder.paid.setText(String.format("Paid %s", new DateTime(billPaid.date).toString("MMMM d")));
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView paid;

        public ViewHolder(View itemView) {
            super(itemView);
            paid = (TextView) itemView.findViewById(R.id.bill_paid_text);
        }
    }
}

File: app/src/main/java/com/djrausch/billtracker/adapters/PaidDateRecyclerViewAdapter.kt
package com.djrausch.billtracker.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.djrausch.billtracker.R
import com.djrausch.billtracker.models.BillPaid
import io.realm.OrderedRealmCollection
import io.realm.RealmRecyclerViewAdapter
import org.joda.time.DateTime

class PaidDateRecyclerViewAdapter(context: Context, dates: OrderedRealmCollection<BillPaid>) : RealmRecyclerViewAdapter<BillPaid, PaidDateRecyclerViewAdapter.ViewHolder>(context, dates, true) {
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        var billPaid: BillPaid = data[position]

        holder.paid.text = "Paid " + DateTime(billPaid.date).toString("MMMM d")
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.row_paid, parent, false)
        return PaidDateRecyclerViewAdapter.ViewHolder(v)
    }

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var paid: TextView

        init {
            paid = v.findViewById(R.id.bill_paid_text) as TextView
        }
    }

}
